### DISCUSSION TITLE
[unreal]编辑器扩展增强支持

### DISCUSSION THREAD

**Discussion by chexiongsheng** [Tencent/puerts#895] [Ideas]
最近有做了imgui的绑定：https://github.com/puerts/UnrealImGuiPuerts
使用的[UnrealImGui](https://github.com/segross/UnrealImGui)似乎还[不支持Editor](https://github.com/segross/UnrealImGui/issues/70)，不过相信改造下支持也不太难。

**Comment by zero16832** [comment]
[https://github.com/ZhuRong-HomoStation/UEImgui.git](url)
可以看看这个，这个已经弄得差不多了

**Reply by blackplume233** [reply]
我无法打开，请问是权限相关的问题么

**Reply by li5414** [reply]
[https://github.com/ZhuRong-HomoStation/UEImgui](https://github.com/ZhuRong-HomoStation/UEImgui)
超链接配置问题

**Reply by chexiongsheng** [reply]
试了下确实不错

**Comment by blackplume233** [comment]
我在使用Puerts + UMG来进行编辑器UI的开发，可以满足大部分的UI需求，但有以下几类需求比较难满足，个人感觉可能也是各种UE编辑器扩展方案共同的难点：
1. 创建独立的面板比较简单，但是要在UE引擎上加各种按钮还是比较麻烦，在右键菜单加按钮和在主界面加按钮及在其他子界面加按钮的逻辑各不相同。 不过这可能是引擎设计上的原因不太好封装，感觉可以讨论一下。
2. 各种DetailLayout或者资产面板还是比较难通过脚本的形式去注册，目前没有想到特别好的思路，可以讨论一下

**Reply by chexiongsheng** [reply]
我觉得像imgui这种其实更适合编辑器扩展（或者gm）中制作UI。它是程序向的，通过代码就能完成UI的绘制，不需要另外搞个可视化编辑环境去编辑UI。

加按钮可以参考下现在的一些python方案，按那样封装下给ts调用。

**Reply by blackplume233** [reply]
是的，单纯从扩展的角度来看我也认为IMGui的实用性更强，使用UMG来扩展编辑器UI开发会有很多很麻烦的问题，包括引擎版本变动，资产顺坏，批量修改等需求都很难满足。 我们是因为有特殊的需求所以采用了UMG来做。
加按钮这套应该是比较好解决，Detail的扩展就有点麻烦 我没太有思路 不知道能不能搞

**Reply by chexiongsheng** [reply]
可以预览下该插件：https://github.com/puerts/EasyEditorPlugin
